docs/development-environment.md

```diff
@@ -1,298 +1,444 @@
+		
+		Most Linux distributions may have an older version of Composer as an installable package, but installing from the official source ensures you have the most up to date version.
+		Note that [we recommend using the Windows Subsystem for Linux](#developing-and-contributing-code-to-jetpack-from-a-windows-machine) to run Composer and PHP.
 
-## Unit-testing
 
-Jetpack includes several [unit tests](https://github.com/Automattic/jetpack/tree/master/tests) that you can run in your local environment before submitting a new Pull Request.
+<a name="setup-development-tools-phpunit"></a>
 
-To get started, you can follow the instructions [here](https://phpunit.de/getting-started.html) to install PHPUnit on your machine. If you are running a recent version of [VVV](https://github.com/Varying-Vagrant-Vagrants/VVV) then Jetpack will automatically detect your wordpress-develop install and you can just run `phpunit` directly.
+* ### PHPUnit
 
-Otherwise you'll need to manually install the `wordpress-develop` branch, as follows:
+	PHPUnit is the unit test framework we use in Jetpack. You can install it by [visitng the official project web site](https://phpunit.de/) and follow the installation instructions there. 
 
-```
-svn co https://develop.svn.wordpress.org/trunk/ /tmp/wordpress-develop
-cd /tmp/wordpress-develop
-cp wp-tests-config-sample.php wp-tests-config.php
-```
+<a name="jetpack-development"></a>
+# Development workflow
 
-Set the database information for your testing DB in the file `/tmp/wordpress-develop/wp-tests-config.php`. You may need to create this database.
+To start work on the Jetpack plugin you need to follow these steps:
 
-To run tests on your machine, you can run `phpunit` while in the Jetpack directory.
+1. [Clone the repository](#jetpack-development-repository)
+2. [Install the development tools](#setup-development-tools)
+3. Make sure Jetpack is enabled on your WordPress site
+4. [Build Jetpack](#jetpack-development-build)
+5. Open `/wp-admin/admin.php?page=jetpack` in your browser.
 
-To run WooCommerce integration tests, you'll need the WooCommerce plugin installed alongside Jetpack (in `../woocommerce`), and you can run:
+<a name="jetpack-development-repository"></a>
+## Clone the repository
 
+Make sure you have `git`, `node`, `yarn`, and a working WordPress installation.
+Clone this repository inside your Plugins directory.
+	
+```sh
+git clone git@github.com:Automattic/jetpack.git
+cd jetpack
 ```
-JETPACK_TEST_WOOCOMMERCE=1 phpunit
-```
-
-To run multisite tests, run:
+	
+ You'll need to have a public SSH key setup with GitHub, which is more secure than saving your password in your keychain.
+ There are more details about [setting up a public key on GitHub.com](https://help.github.com/en/articles/adding-a-new-ssh-key-to-your-github-account).
 
-```
-phpunit -c tests/php.multisite.xml
-```
+<a name="jetpack-development-build"></a>
+## Building Jetpack
```

Comment:
I wonder if we should take the opportunity to split things up a bit here. We now have a build for extensions as well (we could link to https://github.com/Automattic/jetpack/blob/90a1b683c84b87a401ae707599842fe59255ab9b/extensions/README.md ), and a build for Instant Search ( https://github.com/Automattic/jetpack/blob/9aee29df9f2b1ad9031eabe5d7118ad9e5a6f73c/modules/search/instant-search/README.md )

Maybe we can mention that here?

Reply:
Thanks! Added a section about them.